pennylane/devices/qubit/apply_operation.py

```diff
@@ -305,34 +305,34 @@ def apply_mid_measure(
     interface = qml.math.get_deep_interface(state)
     if prng_key is not None:
         # pylint: disable=import-outside-toplevel
-        import jax
+        from jax.random import binomial
+
+        def binomial_fn(n, p):
+            return binomial(prng_key, n, p).astype(int)
 
-        _, key = jax.random.split(prng_key)
-        sample = jax.random.choice(
-            key, np.array([0, 1]), shape=(1,), p=jax.numpy.array([prob0, 1.0 - prob0])
-        )[0]
     else:
-        if rng is None:
-            sampling_fun = np.random.choice
-        else:
-            sampling_fun = rng.choice
-        sample = sampling_fun([0, 1], 1, p=[prob0, 1.0 - prob0])[0]
+        binomial_fn = np.random.binomial if rng is None else rng.binomial
+    sample = binomial_fn(1, 1 - prob0)
     mid_measurements[op] = sample
 
     # Using apply_operation makes it easier to work with JAX
     matrix = qml.math.array([[(sample + 1) % 2, 0.0], [0.0, (sample) % 2]], like=interface)
     state = apply_operation(
+        # qml.Projector([sample], wire), # JAX doesn't like that
```

Comment:
Could you add a note here about why we're using a projector matrix rather than directly applying `qml.Projector`?

```suggestion
```

Reply:
Will do.